File: mcp-code-intelligence-python/src/mcp_code_intel/memory/tier_boost.py
"""TierBoost — tier-based score multiplier in search results."""

from __future__ import annotations

_FACTORS: dict[str, float] = {
    "SEMANTIC": 1.5,
    "PROCEDURAL": 1.3,
    "EPISODIC": 1.1,
    "WORKING": 1.0,
}


def factor(tier: str | None) -> float:
    """Get boost factor for a tier. Higher = ranked higher in results."""
    return _FACTORS.get(tier or "", 1.0)

File: backend/mcp-code-intelligence-kotlin/src/main/kotlin/com/codeintel/memory/search/TierBoost.kt
/** Tier-based score multiplier — SEMANTIC entries rank higher than WORKING. */
package com.codeintel.memory.search

object TierBoost {
    private val FACTORS = mapOf(
        "SEMANTIC" to 1.5,
        "PROCEDURAL" to 1.3,
        "EPISODIC" to 1.1,
        "WORKING" to 1.0
    )

    /** Get boost factor for a tier. Higher = ranked higher in results. */
    fun factor(tier: String?): Double = FACTORS[tier] ?: 1.0
}
